src/VisualStudio/Core/Def/Experimentation/VisualStudioExperimentationService.cs

```diff
@@ -20,20 +20,31 @@ internal class VisualStudioExperimentationService : ForegroundThreadAffinitizedO
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public VisualStudioExperimentationService(IThreadingContext threadingContext, SVsServiceProvider serviceProvider)
-            : base(threadingContext, assertIsForeground: true)
+            : base(threadingContext)
         {
-            try
+            object experimentationServiceOpt = null;
+            MethodInfo isCachedFlightEnabledInfo = null;
+
+            threadingContext.JoinableTaskFactory.Run(async () =>
             {
-                _experimentationServiceOpt = serviceProvider.GetService(typeof(SVsExperimentationService));
-                if (_experimentationServiceOpt != null)
+                await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                try
                 {
-                    _isCachedFlightEnabledInfo = _experimentationServiceOpt.GetType().GetMethod(
-                        "IsCachedFlightEnabled", BindingFlags.Public | BindingFlags.Instance);
+                    experimentationServiceOpt = serviceProvider.GetService(typeof(SVsExperimentationService));
+                    if (experimentationServiceOpt != null)
+                    {
+                        isCachedFlightEnabledInfo = experimentationServiceOpt.GetType().GetMethod(
+                            "IsCachedFlightEnabled", BindingFlags.Public | BindingFlags.Instance);
+                    }
                 }
-            }
-            catch
-            {
-            }
+                catch
+                {
+                }
+            });
+
+            _experimentationServiceOpt = experimentationServiceOpt;
+            _isCachedFlightEnabledInfo = isCachedFlightEnabledInfo;
         }
 
         public bool IsExperimentEnabled(string experimentName)
```

Comment:
it looks like this is the only entrypoint.  can we just make this async?  i walked back from most of hte callers of this, and it seems like they shouldn't have a problem with this being async.

Reply:
Note: if that's unpalatable for v16.  I can understand keeping things sync.  But it seems like we should just make this an async interface for master.